src/spell/modules/spell/spell/src/kv/primitive.rs

```diff
@@ -173,4 +195,46 @@ mod tests {
         let get = spell.get_u32(key);
         assert_eq!(get.num, num * 2, "get_u32 failed: {}", get.error);
     }
+
+    #[marine_test(
+        config_path = "../../tests_artifacts/Config.toml",
+        modules_dir = "../../tests_artifacts"
+    )]
+    fn test_exists(spell: marine_test_env::spell::ModuleInterface) {
+        let key = "num".to_string();
+
+        // check if exists before insertion
+        let exists = spell.exists(key.clone());
+        assert!(exists.success, "first exists failed: {}", exists.error);
+        assert!(!exists.flag, "value exists before set");
+
+        // insert
+        let num = 123;
+        let set = spell.set_u32(key.clone(), num);
+        assert!(set.success, "set_u32 failed: {}", set.error);
+
+        // check if exists after insertion
+        let exists = spell.exists(key.clone());
+        assert!(exists.success, "second exists failed: {}", exists.error);
+        assert!(exists.flag, "value doesn't exists after set_u32");
+
+        // remove
+        let remove = spell.remove_key(key.clone());
+        assert!(remove.success, "remove failed: {}", remove.error);
+
+        // check if exists after remove
+        let exists = spell.exists(key.clone());
+        assert!(exists.success, "third exists failed: {}", exists.error);
+        assert!(!exists.flag, "value still exists after remove_key");
+    }
+
+    #[marine_test(
```

Comment:
Maybe it's time to define these paths in build.rs?

Reply:
How to do that?

Reply:
https://fluence.dev/docs/marine-book/marine-rust-sdk/testing-and-debugging/using-cargo-build-scripts#vs-code